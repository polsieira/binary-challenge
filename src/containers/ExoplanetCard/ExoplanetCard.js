import React from 'react';
import { NavLink } from 'react-router-dom';
import './ExoplanetCard.scss';

const ExoplanetCard = ({ planetName, hostStarName, orbitalPeriod, distance, discoveryYear }) => {
  return (
    <NavLink
      className='exoplanet-link' id={planetName}
      to={`exoplanets/${planetName}`}>
      <div className='ExoplanetCard'>
        <p className="planet-name">{planetName}</p>
        <p className="star-name">{hostStarName}</p>
        <p className="orbital-period">{`${orbitalPeriod} days`}</p>
        <p className="distance">{`${distance} lightyears`}</p>
        <p className="discovery-year">{`${discoveryYear}`}</p>
      </div>
    </NavLink >
  )

}

export default ExoplanetCard;
