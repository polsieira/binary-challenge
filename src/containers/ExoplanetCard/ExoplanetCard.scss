.exoplanet-link {
  color: black;
  text-decoration: none;
}

.ExoplanetCard {
  border-bottom: grey solid 1px;
  cursor: pointer;
  display: flex;
  p {
    width: 20%;
  }

  &:hover {
    background-color: grey;
  }
}