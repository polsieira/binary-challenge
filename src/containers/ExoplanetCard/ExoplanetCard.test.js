import React from 'react';
import { shallow } from 'enzyme';
import ExoplanetCard from './ExoplanetCard';

describe('ExoplanetCard', () => {
  describe('ExoplanetCard component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<ExoplanetCard
        planetName='Earf'
        hostStarName='Suuun'
        orbitalPeriod={365}
        distance={10}
        discoveryYear={2000}
      />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

});