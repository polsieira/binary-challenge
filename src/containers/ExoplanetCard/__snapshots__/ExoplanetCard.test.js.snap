// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExoplanetCard ExoplanetCard component should match the snapshot 1`] = `
<NavLink
  className="exoplanet-link"
  id="Earf"
  to="exoplanets/Earf"
>
  <div
    className="ExoplanetCard"
  >
    <p
      className="planet-name"
    >
      Earf
    </p>
    <p
      className="star-name"
    >
      Suuun
    </p>
    <p
      className="orbital-period"
    >
      365 days
    </p>
    <p
      className="distance"
    >
      10 lightyears
    </p>
    <p
      className="discovery-year"
    >
      2000
    </p>
  </div>
</NavLink>
`;
